src/ai/backend/manager/models/minilang/queryfilter.py

```diff
@@ -0,0 +1,186 @@
+from lark import Lark
+import lark
+from .grammer_checker import checkBracket, checkQuotes, checkEscapeString
+from sqlalchemy.sql import text
+
+
+class Query2sql:
+    __JSON_GRAMMER = r"""
+        ?start: value
+        ?value: query
+              | object
+              | array
+              | filter
+              | string
+              | pair
+              | word
+              | number
+              | "true"             -> true
+              | "false"            -> false
+              | "null"             -> null
+        query  : [word|string] "(" filter ")" object
+        array  : "[" [value ("," value)*] "]"
+        object : "{" [value ("," value)*] "}"
+        filter : [word|string] ":" object
+        pair   : [word|string] ":" [word|string|number]
+        word   : CNAME
+        string : ESCAPED_STRING
+        number : SIGNED_NUMBER
+        %import common.CNAME
+        %import common.ESCAPED_STRING
+        %import common.SIGNED_NUMBER
+        %import common.WS
+        %ignore WS
+    """
+
+    def __init__(self):
+        self._FILTERS = {
+            'and': 'AND',
+            'or': 'OR',
+            'gt': '>',
+            'gte': '>=',
+            'lt': '<',
+            'lte': '<=',
+            'eq': '=',
+            'ne': '!='
+        }
+        self._JSON_PARSER = Lark(
+            Query2sql.__JSON_GRAMMER, parser='lalr', maybe_placeholders=False)
+
+    def query2tree(self, query):
+        if not checkBracket(query):
+            raise ValueError(r'''
+            Check the brackets in the query string.
+            ''')
+        if not checkQuotes(query):
+            raise ValueError(r'''
+            Check the quotes in the value string.
+            If you want to write quotes in the value string, use ' instead of ".
+            ''')
+        if not checkEscapeString(query):
+            raise ValueError(r'''
+            Be careful with the use of Escape string.
+            Escape string can only be used within value string.
+            Double Quotes(") can only be used to represent value string type.
+            If you want to write quotes in the value string, use ' instead of ".
+            ''')
```

Comment:
My expectation is that we should be able to have `\"` in `"`-strings and `\'` in `'`-strings, not only `"` in `'` strings and `'` in `"`-strings.